lxd/events/connections.go

```diff
@@ -128,3 +132,103 @@ func (e *websockListenerConnection) WriteJSON(event any) error {
 
 	return e.Conn.WriteJSON(event)
 }
+
+// NewStreamListenerConnection returns a new http stream listener connection.
+func NewStreamListenerConnection(connection net.Conn) EventListenerConnection {
+	// Send HTTP response to let the client know what to expect.
+	// This is only sent once, and is followed by events.
+	//
+	// The X-Content-Type-Options response HTTP header is a marker used by the server to indicate
+	// that the MIME types advertised in the Content-Type headers should be followed and not be
+	// changed. The header allows you to avoid MIME type sniffing by saying that the MIME types are
+	// deliberately configured.
+	io.WriteString(connection, `HTTP/1.1 200 OK
+Connection: keep-alive
+Content-Type: application/json
+X-Content-Type-Options: nosniff
+
+`)
+
+	return &streamListenerConnection{
+		Conn: connection,
+	}
+}
+
+func (e *streamListenerConnection) Reader(ctx context.Context, recvFunc EventHandler) {
+	ctx, cancelFunc := context.WithCancel(ctx)
+
+	close := func() {
+		e.lock.Lock()
+		defer e.lock.Unlock()
+
+		if ctx.Err() != nil {
+			return
+		}
+
+		e.Close()
+		cancelFunc()
+	}
+
+	defer close()
+
+	pingInterval := time.Second * 10
+
+	// Start reader from client.
+	go func() {
+		defer close()
+
+		buf := make([]byte, 1)
+
+		for {
```

Comment:
I don't understand why this for loop is needed (and the associated `pingInternal` variable), as I understand it there is no out of band ping message occurring (because the client is just a long polling request), and thus we should be able to just have a single read (with no read deadline) that blocks until the client either sends data (invalid) or disconnects.